onnx/test/cpp/data_propagation_test.cc

```diff
@@ -131,23 +129,21 @@ agraph (int32[2,5] x) => (int32[2,5] w)
 {
     y = Shape(x)
     z = Squeeze(y)
-    w = Cast(z)
+    w = Cast<to = 7>(z)
 }
 )ONNX";
-
   RunDataPropagationTest(code);
 }
 
 TEST(DataPropagationImplTest, UnsqueezeTest) {
   const char* code = R"ONNX(
-agraph (int32[2,5] x) => (int32[2,5] w)
+agraph (int32[2,5] x, int32[1] y) => (int32[2,5] t)
 {
-    y = Shape(x)
-    z = Unsqueeze(y)
-    w = Cast(z)
+    z = Shape(x)
+    w = Unsqueeze(z, y)
```

Comment:
I think we can make y a constant as ```y = Constant <...>()``` to test this out.

Reply:
Of course, that would require a data-propagation for Constant. May be that is the issue (until we allow initializers in the parser).

Reply:
Added. Since the data propagation of Unsqueeze won't use axes, not sure whether Constant works as expected in this case.

> Of course, that would require a data-propagation for Constant. May be that is the issue (until we allow initializers in the parser).

Actually I think we can just skip Constant for data propagation? Constant data can still go into inputDataByName for other op's data propagation function to use, right?

Reply:
You are right, my mistake, sorry. Please ignore my comment.

Reply:
I mean it is not required for Unsqueeze